### PR TITLE
Add `didStartPurchaseWithPackage`

### PR DESCRIPTION
**Pull request by vegaro**
Needed for https://github.com/RevenueCat/react-native-purchases/pull/911

### CODE REVIEW

File: ios/PurchasesHybridCommon/PurchasesHybridCommon/Package+HybridAdditions.swift [hunk 1]
```diff
@@ -9,8 +9,12 @@
 import Foundation
 import RevenueCat
 
-internal extension Package {
+public extension Package {
```
Comment (tonidero) [inline ios/PurchasesHybridCommon/PurchasesHybridCommon/Package+HybridAdditions.swift:12]: Do we need to access these from the hybrids now?

**Review by tonidero — APPROVED** [review]

File: ios/PurchasesHybridCommon/PurchasesHybridCommon/Package+HybridAdditions.swift (hunk 1 above)
Comment (vegaro) [reply ios/PurchasesHybridCommon/PurchasesHybridCommon/Package+HybridAdditions.swift:12]: It's from `PurchasesHybridCommonUI` https://github.com/RevenueCat/purchases-hybrid-common/pull/735/files#diff-c673e3957e7bea968ef2fa452b8c46a98aac22c132cf33e54ab4ba6bbb02f025R200

Comment (vegaro) [reply ios/PurchasesHybridCommon/PurchasesHybridCommon/Package+HybridAdditions.swift:12]: for example, CustomerInfo and Offerings are also public for the same reason :)

Comment (tonidero) [reply ios/PurchasesHybridCommon/PurchasesHybridCommon/Package+HybridAdditions.swift:12]: Ohh right, I forgot 😅 Makes sense, thanks!

**Pull request merged**